sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,68 @@
+#[macro_use]
+extern crate log;
+
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+use serde::{Serialize};
+
+// FIXME Get from configuration in the future
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all="PascalCase")]
+struct ECSConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cluster: Option<String>,
+}
+
+fn main() {
+    // Get all settings values for config file templates
+    debug!("Requesting settings values");
+    let settings = match schnauzer::get_settings(&DEFAULT_API_SOCKET) {
+        Ok(s) => s,
+        Err(e) => {
+            error!("Failed to read settings: {}", e);
+            return
```

Comment:
Other settings appliers exit with a non-zero status when they encounter an error. So, you'd do something like:

```suggestion
            std::process::exit(1)
```

Reply:
I saw that some of the other programs treat the `main` fn as a [wrapper around another fn](https://github.com/bottlerocket-os/bottlerocket/blob/7303622bce4ba7f4c1587a0abf9c4a18ffb9db8f/sources/api/schnauzer/src/main.rs#L171-L176).  I assume that's the pattern I'd want to follow?  I haven't touched snafu yet though.

Reply:
Yep, I [suggested this below](https://github.com/bottlerocket-os/bottlerocket/pull/946#discussion_r445074568) and forgot to mention it here. :facepalm: 

The wrapper would be responsible for exiting with the appropriate exit code. For what its worth: you don't *have to* use Snafu, you can just as easily use a String as your "error" type to be returned to your main: `Result<(), String>`. Though, your error type should really be one that actually implements `Error`.  

Aside from the error types, I think there's a set of pre-defined codes that are handled especially but can't find where at the moment. If you need special handling, other than pass/fail, then we might want to seek that out.

Reply:
Tackling this with `snafu` though should be straightforward. This differs from Go where errors are ad-hoc - with `snafu` (and many others like it) you end up with well defined Error types. `snafu` uses enums and its variants that are used to derive "helper" types and conversion implementations.

To get started with `snafu`, I'd suggest starting out by mimicking [`pluto`](https://github.com/bottlerocket-os/bottlerocket/blob/f1c1d2379c0f8f49b2d20090e5b6f23c38cb6d5d/sources/api/pluto/src/main.rs#L48-L69). You need a `mod error` to contain the helper types and your primary `Error` enum (called [`PlutoError` there](https://github.com/bottlerocket-os/bottlerocket/blob/f1c1d2379c0f8f49b2d20090e5b6f23c38cb6d5d/sources/api/pluto/src/main.rs#L63)). Then you can use error helper macros and trait extensions that make working with `?` much more friendly (eg: with `context()` and `ensure!()`).

Reply:
I'm still eagerly awaiting [`std::process::Termination`](https://doc.rust-lang.org/std/process/trait.Termination.html) which can be used to plumb errors all the way to main so that you can say:

```rust
fn main() -> Result<(), ProgramError> {
  // program here, with liberal use of `?`
  Ok(())
}

Reply:
Done!